app/scripts/modules/amazon/src/function/CreateLambdaFunction.tsx

```diff
@@ -0,0 +1,244 @@
+import * as React from 'react';
+import { cloneDeep } from 'lodash';
+import {
+  IFunctionModalProps,
+  FunctionWriter,
+  ReactInjector,
+  ReactModal,
+  TaskMonitor,
+  WizardModal,
+  WizardPage,
+  noop,
+  Application,
+} from '@spinnaker/core';
+
+import { IAmazonFunction, IAmazonFunctionUpsertCommand } from 'amazon/domain';
+import { FunctionBasicInformation } from './configure/FunctionBasicInformation';
+import { ExecutionRole } from './configure/ExecutionRole';
+import { FunctionSettings } from './configure/FunctionSettings';
+import { FunctionEnvironmentVariables } from './configure/FunctionEnvironmentVariables';
+import { Network } from './configure/Network';
+import { AwsFunctionTransformer } from './function.transformer';
+import { FunctionTags } from './configure/FunctionTags';
+import { FunctionDebugAndErrorHandling } from './configure/FunctionDebugAndErrorHandling';
+
+export interface IAmazonCreateFunctionProps extends IFunctionModalProps {
+  app: Application;
+  functionDef: IAmazonFunction;
+}
+export interface IAmazonCreateFunctionState {
+  isNew: boolean;
+  functionCommand: IAmazonFunctionUpsertCommand;
+  taskMonitor: TaskMonitor;
+}
+
+export class CreateLambdaFunction extends React.Component<IAmazonCreateFunctionProps, IAmazonCreateFunctionState> {
+  public static defaultProps: Partial<IAmazonCreateFunctionProps> = {
+    closeModal: noop,
+    dismissModal: noop,
+  };
+
+  constructor(props: IAmazonCreateFunctionProps) {
+    super(props);
+    const functionTransformer = new AwsFunctionTransformer();
+    const funcCommand = props.command
+      ? (props.command as IAmazonFunctionUpsertCommand) // ejecting from a wizard
```

Comment:
On closer inspection I figured that there are only 2 possibilities for funcCommand. It will either be for creating new function or editing existing function. 
I wasn't sure what `ejecting from wizard` meant and just to be consistent with LoadBalancer
I added the casting. I used `CreateApplicationLoadBalancer` as reference.

I am removing the type casting and the change will appear in the next push.